lib/Dialect/FIRRTL/Transforms/LowerCHIRRTL.cpp

```diff
@@ -358,13 +358,14 @@ void LowerCHIRRTLPass::replaceMem(Operation *cmem, StringRef name,
   // Create the memory.
   ImplicitLocOpBuilder memBuilder(cmem->getLoc(), cmem);
   auto symOp = cast<hw::InnerSymbolOpInterface>(cmem);
+  auto prefixAttr = cmem->getAttrOfType<StringAttr>("prefix");
   auto memory = memBuilder.create<MemOp>(
       resultTypes, readLatency, writeLatency, depth, ruw,
       memBuilder.getArrayAttr(resultNames), name,
       cmem->getAttrOfType<firrtl::NameKindEnumAttr>("nameKind").getValue(),
       annotations, memBuilder.getArrayAttr(portAnnotations),
       symOp.getInnerSymAttr(),
-      cmem->getAttrOfType<firrtl::MemoryInitAttr>("init"), StringAttr());
+      cmem->getAttrOfType<firrtl::MemoryInitAttr>("init"), prefixAttr);
```

Comment:
Nit: the arguments of this create are a mix of two patterns:

1. Single-use variables to improve readability.
2. Inlined expressions as the temporaries would be single-use.

Maybe, let's split the difference and refactor this to use `/*argument=*/` comments?

(I know that I prefer inline and you like outline. 😉) Alternatively, single-use variables for everything. I mostly just don't like the fact that the argument immediately before this is of the exact same form as the new one, but chooses a different format. This can happen in a cleanup or prerequisite.

Reply:
I think this was more of a mistake, from when I was hacking together the initial implementation. I moved it to an inline expression, I think its pretty clear that its just forwarding along the prefix attribute.